code/datums/components/gas_connector.dm

```diff
@@ -0,0 +1,72 @@
+/datum/component/gas_connector
+	//Reference to the spawned pipe
+	var/obj/machinery/atmospherics/components/unary/internal_connector
+
+/datum/component/gas_connector/Initialize(obj/parent_obj, internal_volume)
+	. = ..()
+	internal_connector = new /obj/machinery/atmospherics/components/unary(parent_obj.loc)
+	internal_connector.dir = parent_obj.dir
+	internal_connector.airs[1].volume = internal_volume
+	internal_connector.atmos_init()
+	SSair.start_processing_machine(parent_obj)
+
+/datum/component/gas_connector/Destroy(force, silent)
+	var/datum/gas_mixture/air = internal_connector.airs[1]
+	var/turf/parent_turf = get_turf(parent)
+	parent_turf.assume_air(air)
```

Comment:
I don't know much of atmos code, but is this dumping the gas into the air? 

If so, this doesn't seem appropriate to put here - side effects shouldn't be placed in `Destroy`. 

Perhaps register a `COMSIG_ATOM_DECONSTRUCTED` signal if you want it to dump gas when bashed to bits in game. (`deconstruct` is called when an object is beat to non-existence or tooled apart)

Reply:
agree